pos_margin/hooks.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def _create_column(cr, table_name, column_name, column_type):
+    req = "ALTER TABLE %s ADD COLUMN %s %s" % (
+        table_name, column_name, column_type)
+    cr.execute(req)
+
+
+def pre_init_hook(cr):
+    _logger.info("Compute pos_order_line.purchase_price for existing lines")
+    _create_column(cr, 'pos_order_line', 'purchase_price', 'numeric')
+    cr.execute("""
```

Comment:
This can be performed without subquery (the same as in OpenUpgrade) and without the performance penalty it means. Do you figure out how to do it?

Reply:
Not really, could you help me. I'm really not very good for update request.
Regards.

Reply:
Hi @pedrobaeza. Could we consider to merge this PR as it, and search some optimization. The subquery is not generating a big performance penalty. Otherwise, could you point me how to do ?
regards.

Reply:
ping @pedrobaeza